pytest_rpc/helpers.py

```diff
@@ -439,3 +439,107 @@ def ping_ip_from_utility_container(ip, run_on_host):
+        testinfra.CommandResult: Result of command execution.
+    """
+
+    pre_command = ("lxc-attach "
+                   "-n $(lxc-ls -1 | grep {} | head -n 1) "
+                   "-- bash -c".format(container_type))
+    cmd = "{} '{}'".format(pre_command, command)
+    return run_on_host.run(cmd)
+
+
+def run_on_swift(cmd, run_on_host):
+    """Run the given command on the swift container.
+    Args:
+        cmd (str): Command
+        run_on_host (testinfra.Host): Testinfra host object to execute the
+                                      wrapped command on.
+    Returns:
+        testinfra.CommandResult: Result of command execution.
+    """
+
+    command = (". ~/openrc ; "
+               ". /openstack/venvs/swift-*/bin/activate ; "
+               "{}".format(cmd))
+    return run_on_container(command, 'swift', run_on_host)
+
+
+def parse_swift_recon(recon_out):
+    """Parse swift-recon output into list of lists grouped by the content of
+    the delimited blocks.
+    Args:
+        recon_out (str): CLI output from the `swift-recon` command.
+    Returns:
+        list: List of lists grouped by the content of the delimited blocks
+    Example output from `swift-recon --md5` to be parsed:
+    ===============================================================================
+    --> Starting reconnaissance on 3 hosts (object)
+    ===============================================================================
+    [2018-07-19 15:36:40] Checking ring md5sums
+    3/3 hosts matched, 0 error[s] while checking hosts.
+    ===============================================================================
+    [2018-07-19 15:36:40] Checking swift.conf md5sum
+    3/3 hosts matched, 0 error[s] while checking hosts.
+    ===============================================================================
+    """
+
+    lines = recon_out.splitlines()
+    collection = []
+    data = []
+    inData = False
+    i = 0
+    while i < len(lines):
+        if not inData:
+            if lines[i].startswith('=' * 79):
+                inData = True
+        elif lines[i].startswith('=' * 79):
+            inData = False
+            collection.append(data)
+            data = []
+            # Reset counter to use the delimeter to trigger data collection in
+            # the next pass.
+            i = i - 1
+        else:
+            data.append(lines[i])
+        i = i + 1
+    return collection
+
+
+def parse_swift_ring_builder(ring_builder_output):
+    """Parse the supplied output into a dictionary of swift ring data.
+    Args:
+        ring_builder_output (str): The output from the swift-ring-builder
+                                   command.
+    Returns:
+        dictionary: Swift ring data. Empty dictionary if parse fails.
+    Example data:
+        {'zones': 1.0,
+         'replicas': 3.0,
+         'devices': 9.0,
+         'regions': 1.0,
+         'dispersion': 0.0,
+         'balance': 0.78,
+         'partitions': 256.0}
+    """
```

Comment:
The docstring is a bit squished; can you add some newlines to make it more readable?